js/util/util.js

```diff
@@ -0,0 +1,129 @@
+/** Helper function for parsing hybrid node's text.
+ *  Calls parseOr to for parsing courses that are separated by /
+ *  Example:
+ *  const result = parseAnd('CSC108, CSC148, MAT135/136')
+ *  result === [['CSC108', 'CSC148', ['MAT135', 'MAT136']], '']
+ * @param {string} s a combination of course codes
+ * @returns {Array} an array containing parsed course codes
+ */
+export function parseAnd(s) {
+  // input: CSC301/317/318/384/417/419
+  "use strict"
+  var curr = s
+  var andList = []
+  while (curr.length > 0) {
+    if (curr.charAt(0) === "," || curr.charAt(0) === ";" || curr.charAt(0) === " ") {
+      curr = curr.substr(1)
+    } else {
+      var result = parseOr(curr)
+      // result == ['CSC301', 'CSC317', 'CSC318', 'CSC384', 'CSC417', 'CSC419', '']
+      if (curr === result[1]) {
+        console.error("Parsing failed for " + s + "  with curr = " + curr)
+        break
+      } else {
+        curr = result[1]
+        andList.push(result[0])
+      }
+    }
+  }
+  return [andList, curr]
+}
+
+/**
+ * Helper function for parsing hybrid node's text
```

Comment:
It's great to see you adding more documentation! Please make sure all sentences in documentation end with a period (and in general have correct grammar & spelling).